docs/source/java/dataset.rst

```diff
@@ -228,3 +249,50 @@ native objects after using. For example:
     AutoCloseables.close(factory, dataset, scanner);
 
 If user forgets to close them then native object leakage might be caused.
+
+Development Guidelines
+======================
+
+* Related to the note about ScanOptions batchSize argument: Let's try to read a Parquet file with gzip compression and 3 row groups:
+
+    .. code-block::
+
+       # Let configure ScanOptions as:
+       ScanOptions options = new ScanOptions(/*batchSize*/ 32768);
+
+       $ parquet-tools meta data4_3rg_gzip.parquet
+       file schema: schema
+       age:         OPTIONAL INT64 R:0 D:1
+       name:        OPTIONAL BINARY L:STRING R:0 D:1
+       row group 1: RC:4 TS:182 OFFSET:4
+       row group 2: RC:4 TS:190 OFFSET:420
+       row group 3: RC:3 TS:179 OFFSET:838
+
+    In this case, we are configuring ScanOptions batchSize argument equals to
+    32768 rows, it's greater than 04 rows used on the file, then 04 rows is
+    used on the program execution instead of 32768 rows requested.
+
+* Arrow Java Dataset offer native functionalities consuming native artifacts such as:
```

Comment:
I don't understand what this section is telling me. It's a pretty big context switch to go from configuring scan options to building jars. Maybe additional description would be helpful

Reply:
Yes, this is more low level detail, let me delete that part because that ares JNI libs running under the hood.